go/oasis-test-runner/oasis/compute.go

```diff
@@ -230,6 +233,22 @@ func (net *Network) NewCompute(cfg *ComputeCfg) (*Compute, error) {
 	if cfg.StorageBackend == "" {
 		cfg.StorageBackend = database.BackendNameBadgerDB
 	}
+	// Initialize runtime state paths.
+	for i, path := range cfg.RuntimeStatePaths {
+		if path == "" {
+			continue
+		}
+		stateDir := registry.GetRuntimeStateDir(host.DataDir(), net.Runtimes()[i].descriptor.ID)
+		if err = os.MkdirAll(stateDir, os.ModePerm); err != nil {
+			return nil, fmt.Errorf("oasis/compute: failed creating runtime state dir: %w", err)
+		}
+		cmd := exec.Command("rsync", "-a", path, stateDir)
```

Comment:
Is there a Go-native way to do this instead of having an implicit dependency on rsync?

Should maybe also add -S as these databases use sparse files? Although I'm not sure if this is also needed on the local filesystem.

Reply:
Here are some copy-dir implementations from the wild: [go/internal](https://github.com/golang/go/blob/b2b8872c876201eac2d0707276c6999ff3eb185e/src/cmd/go/internal/modcmd/vendor.go#L390-L422), [moby/moby]( https://github.com/moby/moby/blob/be628115bafce21680343cef30b7051bf8a98b71/daemon/graphdriver/copy/copy.go#L113-L256), [hasura/graphql](https://github.com/hasura/graphql-engine/blob/6532a658eb3dfae9dde251c6c9377f2268ecc8a8/cli/util/copy.go#L13-L115). 

Could add a similar helper function and stick it into some util/helper package.